target/ppc/translate/vsx-impl.c.inc

```diff
@@ -1546,6 +1546,42 @@ static bool trans_LXVKQ(DisasContext *ctx, arg_X_uim5 *a)
     return true;
 }
 
+static bool trans_XVTLSBB(DisasContext *ctx, arg_X_bf_xb *a)
+{
+    TCGv_i64 xb, all_true, all_false, crbf;
+
+    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
+    REQUIRE_VSX(ctx);
+
+    xb = tcg_temp_new_i64();
+    all_true = tcg_const_i64(1);
+    all_false = tcg_const_i64(1);
+    crbf = tcg_temp_new_i64();
+
+    for (int dw = 0; dw < 2; dw++) {
+        get_cpu_vsr(xb, a->xb, dw);
+        for (int nbyte = 0; nbyte < 8; nbyte++) {
```

Comment:
I believe you can get rid of this inner loop by:
- building a 64-bit mask, similar to what @mferst did in #48 
- doing a single and/andc pair to get ``mask & xb`` (or ``mask & ~xb``)
- using TCG's movcond to set all_true/all_false to 0 if and only if the and/andc results above are not equal to the full mask

Reply:
done, thanks

Reply:
This wasn't exactly what I had in mind, which is:

```
static bool trans_XVTLSBB(DisasContext *ctx, arg_X_bf_xb *a)
{
    TCGv_i64 xb, temp, all_true, all_false;
    TCGv_i64 zero, mask;

    REQUIRE_INSNS_FLAGS2(ctx, ISA310);
    REQUIRE_VSX(ctx);

    zero = tcg_constant_i64(0);
    mask = tcg_constant_i64(dup_const(ME_8, 1));

    all_true = tcg_const_i64(0b1000);
    all_false = tcg_const_i64(0b0010);

    xb = tcg_temp_new_i64();
    temp = tcg_temp_new_i64();

    for (int dw = 0; dw < 2; dw++) {
        get_cpu_vsr(xb, a->xb, dw);

        /* test all_true */
        tcg_gen_and_i64(temp, mask, xb);
        tcg_gen_movcond_i64(TCG_COND_EQ, all_true, temp,
                            mask, all_true, zero);

        /* test all_false */
        tcg_gen_andc_i64(temp, mask, xb);
        tcg_gen_movcond_i64(TCG_COND_EQ, all_false, temp,
                            mask, all_false, zero);
    }

    tcg_gen_or_i64(temp, all_true, all_false);
    tcg_gen_extrl_i64_i32(cpu_crf[a->bf], temp);

    tcg_temp_free_i64(xb);
    tcg_temp_free_i64(temp);
    tcg_temp_free_i64(all_true);
    tcg_temp_free_i64(all_false);

    return true;
}
```

Would this work?

For the record, the idea for not truncating that I presented (and later discarded) below would be to define ``all_true``, ``all_false`` and ``zero`` as TCGv_i32. Then I could ``or`` the two together directly into cpu_crf[a->bf]. But movcond isn't defined for 64-bit condition register arguments (``c1`` and ``c2``) and 32-bit ``dest``, ``v1``, ``v2``. In other words:
```
        tcg_gen_movcond_*(TCG_COND_EQ, all_true, temp,
                            mask, all_true, zero);
```
isn't an option if ``all_true`` and ``zero`` are TCGv_i32 and ``temp`` and ``mask`` are TCGv_i64.

So I went with everything TCGv_i64 and kept the truncate later on.

Reply:
Oh, that's a great implementation. I see now what you meant with, it sure works and is way cleaner. 

Thank you very much for your help. Sent a new commit